src/pkg/cli/client/byoc/gcp/byoc.go

```diff
@@ -714,7 +714,14 @@ func (b *ByocGcp) PutConfig(ctx context.Context, req *defangv1.PutConfigRequest)
 	term.Debugf("Creating secret %q", secretId)
 
 	if _, err := b.driver.CreateSecret(ctx, secretId); err != nil {
-		if stat, ok := status.FromError(err); ok && stat.Code() == codes.AlreadyExists {
+		if stat, ok := status.FromError(err); ok && stat.Code() == codes.PermissionDenied {
+			if err := b.driver.EnsureAPIsEnabled(ctx, "secretmanager.googleapis.com"); err != nil {
+				return annotateGcpError(err)
+			}
+			if _, err := b.driver.CreateSecret(ctx, secretId); err != nil {
+				return fmt.Errorf("failed to create secret %q: %w", secretId, err)
+			}
+		} else if stat, ok := status.FromError(err); ok && stat.Code() == codes.AlreadyExists {
```

Comment:
This seems like a more common pattern:
```suggestion
			_, err = b.driver.CreateSecret(ctx, secretId)
		}
		if stat, ok := status.FromError(err); ok && stat.Code() == codes.AlreadyExists {
```